### PR TITLE
Add token to GoalSummary on summary page

### PR DESCRIPTION
**Pull request by mmmmillar**
**What**  
- Pass token to GoalSUmmary on summary page
- Add logging to api fetches

### CODE REVIEW

File: api/api.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
+import logger from 'lib/infrastructure/logging/logger';
 import { HttpStatusError } from './HttpStatusError';
 
 async function request(path, { token, ...options }) {
   const url = `${process.env.NEXT_PUBLIC_API_URL}${path}`;
+  logger.info(`Fetching ${url}`, options?.body ? options.body : null);
```
Comment (t-davies) [inline api/api.js:6]: `options?.body` should produce the same output as `options?.body ? options.body : null` I think?

File: api/api.js [hunk 2]
```diff
@@ -16,6 +18,7 @@ async function request(path, { token, ...options }) {
   if (response.ok) {
     return response.json();
   } else {
+    logger.error(`Fetching ${url} failed`, response);
```
Comment (t-davies) [inline api/api.js:21]: Not 100% sure, but the response object _could_ contain circular references that will cause this to throw `TypeError: Converting circular structure to JSON`. Might be worth just logging the status code maybe?

File: api/api.js (hunk 1 above)
Comment (mmmmillar) [reply api/api.js:6]: ha true

File: api/api.js (hunk 2 above)
Comment (mmmmillar) [reply api/api.js:21]: righto

**Pull request merged**